packages/core/src/shared/plugins/suggestion/SuggestionPlugin.ts

```diff
@@ -382,57 +325,72 @@ export function createSuggestionPlugin<T extends SuggestionItem>({
 
     props: {
       handleKeyDown(view, event) {
-        const { active } = (this as Plugin).getState(view.state);
-
-        if (!active) {
-          // activate the popup on 'char' keypress (e.g. '/')
-          if (event.key === char) {
-            view.dispatch(
-              view.state.tr
-                .insertText(char)
-                .scrollIntoView()
-                .setMeta(pluginKey, { activate: true, type: "slash" })
-            );
-            // return true to cancel the original event, as we insert / ourselves
-            return true;
-          }
-        } else {
-          const { items, range, selectedItemIndex } = pluginKey.getState(
-            view.state
+        // Shows the menu if the default trigger character was pressed.
+        if (event.key === defaultTriggerCharacter) {
```

Comment:
minor detail, I think we should only do this when not active (now, the menu moves to the right when typing multiple slashes)

Reply:
Yeah I think you're right. I originally preferred this as with the current implementation, writing more "/" characters doesn't filter the items and you can do it endlessly. I'll change it so it's more in line with what Notion does, i.e. treat "/" characters as regular characters when the menu is open, so typing many in a row will return no items and eventually close the menu.

Reply:
Also the fade animation is a weird one, but I think I figured it out. The problem is two-fold:

1. We show and hide the React UI but mounting/unmounting the menu components and adding/removing the root element from the DOM, so the component doesn't animate when hiding because it just gets removed from the DOM straight away.
2. The animation is set by Tippy. While we used Tippy before and still use it now, the way we create a Tippy tooltip for the menus to go in is different. Before, we used the `tippy()` function to setup the tooltip for each menu to go in, whereas now we use the `<Tippy />` React component.